### DISCUSSION TITLE
Help integrating Chakra-UI custom radio with RHF

### DISCUSSION THREAD

**Discussion by paambaati** [react-hook-form/react-hook-form#4591] [Q&A] [answered]
X-posting from https://github.com/chakra-ui/chakra-ui/discussions/3724

I'm trying to build a [Chakra-UI custom radio group component](https://chakra-ui.com/docs/form/radio#custom-radio-buttons) that also needs to submit values to RHF v6.

I have a CodeSandbox forked from the [RHF example in the Guides section](https://chakra-ui.com/guides/integrations/with-hook-form), and I'm having trouble getting a custom radio group to register with RHF.

https://codesandbox.io/s/chakra-ui-react-hook-form-custom-radio-2un9h?file=/src/HookForm.js

Has anyone had success getting these 2 woking together?

**Comment by jorisre** [comment, marked as answer]
Hi @paambaati 👋🏻 

You should use `Controller` or `useController`.

Here is a working example with `useController`: https://codesandbox.io/s/chakra-ui-react-hook-form-custom-radio-forked-xve7r

👉🏻  [`useController` doc](https://react-hook-form.com/v6/api#useController)
👉🏻  [`Controller` doc](https://react-hook-form.com/v6/api#Controller)

**Reply by paambaati** [reply]
@jorisre Thanks, this worked perfectly!

**Reply by 4ortytwo** [reply]
@jorisre, can't access the codesandbox, could you please show that again?

**Reply by jorisre** [reply]
@4ortytwo Here is the example using `useController` https://codesandbox.io/s/chakra-ui-react-hook-form-custom-radio-forked-xve7r

**Reply by 4ortytwo** [reply]
Thank you, @jorisre, I posted mine as well.

**Comment by bluebill1049** [comment]
have a look at this example too: https://codesandbox.io/s/chakra-ui-5mp8g

**Comment by 4ortytwo** [comment]
I've got an alternative solution here: https://codesandbox.io/s/react-hook-form-v7-custom-chakra-radio-group-rhc6q 
It uses `useController` as well and doesn't use refs.

**Reply by abdulrahman-mh** [reply]
you solution have some errors, and also a deprecated methods🐛
